### PR TITLE
Add "Security considerations" section to high-quality plugins article

### PR DESCRIPTION
**Pull request by erikadoyle**
Include guidance for URL-based dialogs on preparing for 3rd party cookie deprecation

### CODE REVIEW

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/high-quality-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/99a128e0451d3c2479875b7e6a2956d1bdf5e257/msteams-platform%2Fmessaging-extensions%2Fhigh-quality-message-extension.md) | 91 | 22 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/f9ee3806-66df-47c2-818c-8b9bd51854b3) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/high-quality-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/e85fbe5e3c3067341a0f4a195027f8518fb96086/msteams-platform%2Fmessaging-extensions%2Fhigh-quality-message-extension.md) | 91 | 22 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/98879a54-962e-4bdd-b306-61ff1aa8f5db) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/messaging-extensions/high-quality-message-extension.md [hunk 1]
```diff
@@ -651,6 +651,16 @@ Message extensions respond to a user input with an Adaptive Card. An Adaptive Ca
...
 
+## Security best practices
+
+If your message extension plugin uses URL-based dialogs (task modules), be aware that web browsers are moving toward [deprecating third-party cookies](https://blogs.windows.com/msedgedev/2024/03/05/new-privacy-preserving-ads-api/). Best practice is to prepare now for third-party cookie deprecation by taking the following measures:
+
+1. Move away from third-party cookies.
+
+2. If your app requires cookies, use either [Cookies Having Partitioned State (CHIPS)](https://developer.mozilla.org/docs/Web/Privacy/Privacy_sandbox/Partitioned_cookies) or [Storage Access API](https://developer.mozilla.org/en-US/docs/Web/API/Storage_Access_API) to secure your cookies.
+
+3. Remove any fallbacks you may have put in place for TeamsJS initialization failures. If your URL-based dialog fails upon `app.initialize`, don't fall back to opening it as a regular web app, as this exposes it to third-party cookie attack vectors.
```
Comment (v-ypalikila) [inline msteams-platform/messaging-extensions/high-quality-message-extension.md:662]:
```suggestion
If your message extension plugin uses URL-based dialogs (task modules), it’s important to note Web browsers are increasingly [deprecating third-party cookies](https://blogs.windows.com/msedgedev/2024/03/05/new-privacy-preserving-ads-api/). To ensure your plugin remains functional and secure, consider implementing the following best practices:

1. **Transition from third-party cookies**: Begin phasing out the use of third-party cookies in your application.

1. **Secure cookies**: If your app requires cookies, use either [Cookies Having Partitioned State (CHIPS)](https://developer.mozilla.org/docs/Web/Privacy/Privacy_sandbox/Partitioned_cookies) or [Storage Access API](https://developer.mozilla.org/en-US/docs/Web/API/Storage_Access_API) to secure your cookies. )to secure your cookies.

1. **Remove TeamsJS initialization fallbacks**: Avoid implementing fallback mechanisms for TeamsJS initialization failures. If a URL-based dialog doesn't successfully initialize with `app.initialize`, refrain from defaulting to a standard web app launch as it makes your app vulnerable to third-party cookie attack vectors.
```

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/high-quality-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/01dde3d32341d95571f4b9ab1a1b1e27ccf7ed53/msteams-platform%2Fmessaging-extensions%2Fhigh-quality-message-extension.md) | 93 | 18 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/122cbb82-12b0-47e9-98ec-6be1558af527) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/messaging-extensions/high-quality-message-extension.md [hunk 2]
```diff
@@ -651,6 +651,16 @@ Message extensions respond to a user input with an Adaptive Card. An Adaptive Ca
 * Adaptive Cards must include a URL as part of the [metadata](https://adaptivecards.io/explorer/Metadata.html), which allows cards to be easily copied from one hub to another. [*Recommended*]
 * Apart from thumbnails, any image in an Adaptive Card must have an alt-text. [*Recommended*]
 
+## Security best practices
+
+If your message extension plugin uses URL-based dialogs (task modules), it’s important to note that web browsers are [deprecating third-party cookies](https://blogs.windows.com/msedgedev/2024/03/05/new-privacy-preserving-ads-api/). To ensure your plugin remains functional and secure, request you to implementing the following best practices:
```
Comment (erinha) [inline msteams-platform/messaging-extensions/high-quality-message-extension.md:656]: I think the last clause has a typo: "request you to implementing the following best practices".

Should that be something like "please implement the following best practices"?

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/high-quality-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/cd1ddffab1b7a6195628e5a992a92a88459808e9/msteams-platform%2Fmessaging-extensions%2Fhigh-quality-message-extension.md) | 93 | 17 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/112f3e4a-ab15-4562-8e5f-cec479f42449) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

File: msteams-platform/messaging-extensions/high-quality-message-extension.md (hunk 2 above)
Comment (erikadoyle) [reply msteams-platform/messaging-extensions/high-quality-message-extension.md:656]: Thanks! Updated with suggested.

**Comment by acrolinxatmsft1** [comment]
## Acrolinx Scorecards 
 
**We currently enforce a minimum score of 80.**
 
Click the scorecard links for each article to review the Acrolinx feedback on grammar, spelling, punctuation, writing style, and terminology:
| Article | Score | Issues | Scorecard | Processed |
| ------- | ----- | ------ | --------- | --------- |
| [msteams-platform/messaging-extensions/high-quality-message-extension.md](https://github.com/MicrosoftDocs/msteams-docs/blob/55a7aea8607c8cd474ffb89b721d195d52e77f08/msteams-platform%2Fmessaging-extensions%2Fhigh-quality-message-extension.md) | 93 | 18 | [link](https://microsoft-ce-csi.acrolinx.cloud/api/v1/checking/scorecards/1c4d6a82-2e56-459f-9289-4c3d659b83e3) | [:white_check_mark:](# "Successfully processed") |

**More info about Acrolinx**
 
Use the Acrolinx extension, or sidebar, in Visual Studio Code to check spelling, grammar, style, tone, clarity, and key terminology when you're creating or updating content. For more information, see [Use the Visual Studio Code extension to run Acrolinx locally](https://review.docs.microsoft.com/office-authoring-guide/acrolinx-vscode).

**Review by AkashGoel-MS — APPROVED** [review]

**Review by AbhinavSantoshi — APPROVED** [review]

**Review by AbhinavSantoshi — APPROVED** [review]
Changes looks good to me

**Review by v-ypalikila — APPROVED** [review]
Approved.

**Pull request closed without merging**